File: output/deepseek_r1/codenet/direct_translation/java/python/successful_translation/s173032907.py

n = int(input())
a = list(map(int, input().split()))

dp1 = [0] * n
current1 = 0
cost1 = 0
for i in range(n):
    cost1 += abs(a[i] - current1)
    current1 = a[i]
    dp1[i] = cost1

dp2 = [0] * n
current2 = 0
cost2 = 0
for j in reversed(range(n)):
    cost2 += abs(a[j] - current2)
    current2 = a[j]
    dp2[j] = cost2

for i in range(n):
    if i == 0:
        print(dp2[i + 1] + abs(a[1]))
    elif i == n - 1:
        print(dp1[i - 1] + abs(a[i - 1]))
    else:
        print(dp1[i - 1] + dp2[i + 1] + abs(a[i - 1] - a[i + 1]))

File: dataset/codenet/Java/Code/s173032907.java
import java.util.Scanner;

public class Main implements Runnable {

    private static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        new Thread(null, new Main(), "", 128 * 1024 * 1024).start();
    }

    public void run() {
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        long cost1 = 0;
        long cost2 = 0;
        int current1 = 0;
        int current2 = 0;
        long[] dp1 = new long[n];
        long[] dp2 = new long[n];
        for (int i = 0, j = n - 1; i < n; i++, j--) {
            cost1 += Math.abs(a[i] - current1);
            current1 = a[i];
            dp1[i] = cost1;
            cost2 += Math.abs(a[j] - current2);
            current2 = a[j];
            dp2[j] = cost2;
        }

        for (int i = 0; i < n; i++) {
            if (i == 0) {
                System.out.println(dp2[i + 1] + Math.abs(a[1]));
            } else if (i == n - 1) {
                System.out.println(dp1[i - 1] + Math.abs(a[i - 1]));
            } else {
                System.out.println(dp1[i - 1] + dp2[i + 1] + Math.abs(a[i - 1] - a[i + 1]));
            }
        }
    }
}